pyanaconda/ui/gui/spokes/eula.py

```diff
@@ -0,0 +1,103 @@
+    icon = "application-certificate-symbolic"
+    title = CN_("GUI|Spoke", "_License Information")
+    category = LicensingCategory
+
+    @staticmethod
+    def get_screen_id():
+        """Return a unique id of this UI screen."""
+        return "license-information"
+
+    def initialize(self):
+        log.debug("initializing the EULA spoke")
+        NormalSpoke.initialize(self)
+
+        self._have_eula = True
+        self._eula_buffer = self.builder.get_object("eulaBuffer")
+        self._agree_check_button = self.builder.get_object("agreeCheckButton")
+        self._agree_label = self._agree_check_button.get_child()
+        self._agree_text = self._agree_label.get_text()
+
+        log.debug("looking for the license file")
+        license_file = eula.get_license_file_name()
+        if not license_file:
+            log.error("no license found")
+            self._have_eula = False
+            self._eula_buffer.set_text(_("No license found"))
+            return
+
+        self._eula_buffer.set_text("")
+        itr = self._eula_buffer.get_iter_at_offset(0)
+        log.debug("opening the license file")
+        with open(license_file, "r") as fobj:
+            # insert the first line without prefixing with space
+            try:
+                first_line = next(fobj)
+            except StopIteration:
+                # nothing in the file
+                return
+            self._eula_buffer.insert(itr, first_line.strip())
+
+            # EULA file may be preformatted for the console, we want to let Gtk
+            # format it (blank lines should be preserved)
+            for line in fobj:
+                stripped_line = line.strip()
+                if stripped_line:
+                    self._eula_buffer.insert(itr, " " + stripped_line)
+                else:
+                    self._eula_buffer.insert(itr, "\n\n")
+
+    def refresh(self):
+        self._agree_check_button.set_sensitive(self._have_eula)
+        self._agree_check_button.set_active(self.data.eula.agreed)
+
+    def apply(self):
+        self.data.eula.agreed = self._agree_check_button.get_active()
+
+    @property
+    def completed(self):
+        return not self._have_eula or self.data.eula.agreed
+
+    @property
+    def status(self):
+        if not self._have_eula:
+            return _("No license found")
+
+        return _("License accepted") if self.data.eula.agreed else _("License not accepted")
+
+    @classmethod
+    def should_run(cls, environment, data):
+        if eula.eula_available():
+            # don't run if we are in initial-setup in reconfig mode and the EULA has already been accepted
+            if FirstbootOnlySpokeMixIn.should_run(environment, data) and data and data.firstboot.firstboot == FIRSTBOOT_RECONFIG and data.eula.agreed:
+                log.debug("not running license spoke: reconfig mode & license already accepted")
+                return False
+            return True
+        return False
```

Comment:
This needs changing a bit. All anaconda spokes work such that they are always visible, and kickstart changes only if they are completed or not.
- Since the spoke is not reused by initial setup, I think you can leave out the `FIRSTBOOT_RECONFIG` part. @M4rtinK - is that correct please?
- The spoke visibility in `should_run` should depend only on eula presence.
- The `data.eula.agreed` bit should set up the checkbox, I think in `initialize`.

Reply:
> The data.eula.agreed bit should set up the checkbox, I think in initialize

Do you mean running GUI/TUI with a kickstart with `eula --agreed`? What is the use case for this, who and why will do that? Is it possible without hackery?

Reply:
Yes, it's actually very easy. If you don't make anaconda stop on incomplete kickstart, you can use a "partial kickstart" that does not satisfy all needed inputs, and the UIs will be pre-filled when it stops on the hub. It's a thing that is not really nice to support, but invaluable for testing. I run 99% of my anaconda debugging with this, so that I don't have to set it all every time I retry my changes :P Some spins also use this, except hidden in the "default kickstart", together with hiding spokes in config, to set up things that users can't change. It's a bit of a dark art, really.

The rest of the UI works this way, so it would be best to make it consistent.

Reply:
>Since the spoke is not reused by initial setup, I think you can leave out the FIRSTBOOT_RECONFIG part. @M4rtinK - is that correct please?

Yes, I think it really makes more sense to just add the ability to Anaconda to show the EULA without also trying to reuse that screen from Initial Setup. This nicely shows the benefits of that - we can drop the Initial setup specific mixin (which still refers to the old "Firstboot" tool, to be even more confusing) as well as the Initial Setup specific reconfig functionality. 

As for when the EULA spoke should show up - I think we should just check if an EULA is in the expected place & show it if there is something. That's how it works for Initial Setup & should work for the installer as well.